### ISSUE TITLE
Add support for new item type "Data Pipelines"

### ISSUE THREAD

**Issue by chris-fox** [Esri/solution.js#1082]
This is a simple json item type that can have dependencies on feature service items. Below is an example of the json data for the item:

```
{
	"inputs": [{
		"id": "1878f651b131",
		"type": "FeatureServiceSource",
		"parameters": {
			"layer": {
				"value": {
					"itemId": "fe713ca8996a4db988b8b4f2e5949616",
					"layerId": 0,
					"uiTitle": "Requests"
				}
			}
		}
	}],
	"tools": [{
		"id": "1878f811c692",
		"type": "FilterByAttributeTool",
		"parameters": {
			"uiLanguage": {
				"value": "arcade"
			},
			"arcadeExpression": {
				"value": "$record.reqcategory == 'Blight'"
			},
			"input": {
				"fromId": "1878f651b131"
			}
		}
	}, {
		"id": "1878f81da253",
		"type": "RemoveDuplicatesTool",
		"parameters": {
			"keyFields": {
				"value": ["OBJECTID"]
			},
			"input": {
				"fromId": "1878f811c692"
			}
		}
	}],
	"outputs": [{
		"id": "1878f828e1c4",
		"type": "FeatureServiceSink",
		"parameters": {
			"outputMethod": {
				"value": "replace"
			},
			"layer": {
				"value": {
					"itemId": "5fe495577b434f0e865d3b288b9fcb73",
					"layerId": 0,
					"uiTitle": "New Layer"
				}
			},
			"replaceMethod": {
				"value": "truncateAppend"
			},
			"geometryField": {
				"value": "shape"
			},
			"input": {
				"fromId": "1878f81da253"
			}
		}
	}]
}
```

In the inputs array we should look for types of "FeatureServiceSource" and use the itemId property to find the feature layer and templatize it. In the outputs array we should look for types of "FeatureServiceSink" and use the itemId property to find the feature layer and templatize it.

**Comment by MikeTschudi** [comment]
Implemented via commit 7d6a7330ad44caf484bb117c6aaaa0eb71620b85

**Comment by shoe913** [comment]
Verified
